### PR TITLE
Fix compile warnings in segline

### PR DESCRIPTION
**Pull request by xezon**

### CODE REVIEW

File: src/w3d/renderer/segline.cpp [hunk 1]
```diff
@@ -73,7 +73,7 @@ void SegmentedLineClass::Set_Point_Location(unsigned int point_idx, const Vector
     Invalidate_Cached_Bounding_Volumes();
 }
 
-void SegmentedLineClass::Get_Point_Location(unsigned int point_idx, Vector3 &loc)
+void SegmentedLineClass::Get_Point_Location(int point_idx, Vector3 &loc)
 {
     if (point_idx < m_pointLocations.Count()) {
```
Comment (xezon) [inline src/w3d/renderer/segline.cpp]: `m_pointLocations` (SimpleDynVecClass<>) is using `int` for indexing. Therefore `unsigned int` in this class will issue compiler warning when interacting with member class.

File: src/w3d/renderer/segline.cpp [hunk 2]
```diff
@@ -388,7 +388,7 @@ float SegmentedLineClass::Get_Post_Increment_Value() const
 
     float poly = (float)Get_Num_Polys();
     float poly2 = poly + poly;
-    float p = (1.0 - 0.5 / (poly2 * poly2));
+    float p = (1.0f - 0.5f / (poly2 * poly2));
```
Comment (xezon) [inline src/w3d/renderer/segline.cpp:391]: Compiler warning here.

**Comment by jonwil** [comment]
We have clear proof that these functions take an unsigned int as a parameter and I would like to keep things matching. The compiler error could be fixed by a cast inside the function.

**Comment by xezon** [comment]
Ok. I reinstated unsigned and added static_cast.

**Review by jonwil — APPROVED** [review]
Looks good to me.

**Pull request merged**